src/client/components/date-range-picker/calendar.mocha.ts

```diff
@@ -0,0 +1,168 @@
+      [date("2019-10-13", tz), date("2019-10-14", tz), date("2019-10-15", tz), date("2019-10-16", tz), date("2019-10-17", tz), date("2019-10-18", tz), date("2019-10-19", tz)],
+      [date("2019-10-20", tz), date("2019-10-21", tz), date("2019-10-22", tz), date("2019-10-23", tz), date("2019-10-24", tz), date("2019-10-25", tz), date("2019-10-26", tz)],
+      [date("2019-10-27", tz), date("2019-10-28", tz), date("2019-10-29", tz), date("2019-10-30", tz), date("2019-10-31", tz)]
+    ];
+
+    it("should calculate for UTC october (no summer time change)", () => {
+      const timezone = Timezone.UTC;
+      const firstOctober2019 = date("2019-10-01", timezone);
+      assertEqualCalendarMatrix(monthToWeeks(firstOctober2019, timezone, getLocale()), october2019Weeks(timezone));
+    });
+
+    it("should calculate for Warsaw october (summer time change)", () => {
+      const timezone = Timezone.fromJS("Europe/Warsaw");
+      const firstOctober2019 = date("2019-10-01", timezone);
+      assertEqualCalendarMatrix(monthToWeeks(firstOctober2019, timezone, getLocale()), october2019Weeks(timezone));
+    });
+  });
+});
+
+describe("calendarDays", () => {
+  it("should calculate calendar for March UTC", () => {
+    const utc = Timezone.UTC;
+    const firstMarch2010 = date("2010-03-01", utc);
+    assertEqualCalendarMatrix(calendarDays(firstMarch2010, utc, getLocale()), [
+      [date("2010-02-28", utc), date("2010-03-01", utc), date("2010-03-02", utc), date("2010-03-03", utc), date("2010-03-04", utc), date("2010-03-05", utc), date("2010-03-06", utc)],
+      [date("2010-03-07", utc), date("2010-03-08", utc), date("2010-03-09", utc), date("2010-03-10", utc), date("2010-03-11", utc), date("2010-03-12", utc), date("2010-03-13", utc)],
+      [date("2010-03-14", utc), date("2010-03-15", utc), date("2010-03-16", utc), date("2010-03-17", utc), date("2010-03-18", utc), date("2010-03-19", utc), date("2010-03-20", utc)],
+      [date("2010-03-21", utc), date("2010-03-22", utc), date("2010-03-23", utc), date("2010-03-24", utc), date("2010-03-25", utc), date("2010-03-26", utc), date("2010-03-27", utc)],
+      [date("2010-03-28", utc), date("2010-03-29", utc), date("2010-03-30", utc), date("2010-03-31", utc), date("2010-04-01", utc), date("2010-04-02", utc), date("2010-04-03", utc)]
+    ]);
+  });
+});
+
+describe("shiftOneDay", () => {
+  const utc = Timezone.UTC;
+  const warsaw = Timezone.fromJS("Europe/Warsaw");
+
+  it("shift by one day in UTC", () => {
+    const start = moment("2010-01-03", utc);
+    const expected = moment("2010-01-04", utc);
+    expect(shiftOneDay(start).isSame(expected)).to.be.true;
+  });
+
+  it("shift by one day in Europe/Warsaw", () => {
+    const start = moment("2010-01-03", warsaw);
+    const expected = moment("2010-01-04", warsaw);
+    expect(shiftOneDay(start).isSame(expected)).to.be.true;
+  });
+
+  it("shift by one day in Europe/Warsaw across DST forwards", () => {
```

Comment:
wrap those two in `describe("shift by one day in Europe/Warsaw across DST forwards")` and entitle them as `the day of time change` and `the day after time change` or something like this.